dag-trigger/airflow-dag-trigger.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+# TODO: move this script to pipeline / airflow project
+
+set -e
+
+dag_id=$1
+
+echo "Fetching state for DAG $dag_id"
+
+dag_url="$AIRFLOW_ENDPOINT/api/v1/dags/${dag_id}"
+is_paused=$(curl -sSf -u "$AIRFLOW_USERNAME":"$AIRFLOW_PASSWORD" $dag_url | jq .is_paused)
+
+if [ "$is_paused" == "true" ]; then
+  echo "DAG $dag_id is paused and cannot be run at this time."
+  exit 1
+fi
+
+dag_run_url="$dag_url/dagRuns"
+echo "Triggering DAG run on airflow API at: $dag_run_url"
+
+run_json=$(curl -sSf -u "$AIRFLOW_USERNAME":"$AIRFLOW_PASSWORD" -X POST \
+  "$dag_run_url" \
+  -H 'Cache-Control: no-cache' \
+  -H 'Content-Type: application/json' \
+  -d "{\"conf\": $DAG_CONF}")
```

Comment:
if using `set -u` and `$DAG_CONF` is undefined, this will fail. Should this be a script parameter instead of an env variable, and defined with something like `$dag_conf=${2:-'{}'}` ?

Reply:
Maybe if we did something like `${DAG_CONF:-{}}` ?

Reply:
Or couldn't we do this in `dag-trigger-job.yaml` `{{ .Values.dagConfiguration | default "{}" | quote }}`